src/Aspire.ProjectTemplates/templates/aspire-apphost/Aspire.AppHost1.csproj

```
@@ -1,5 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
+  <Sdk Name="Aspire.AppHost.Sdk" Version="!!REPLACE_WITH_LATEST_VERSION!!" />
```

Why is this not just

```suggestion
<Project Sdk="Aspire.AppHost.Sdk">
```

?

---

We could with some work, but I opted not for several reasons:

- Aspire.AppHost.Sdk doesn't import Microsoft.NEt.Sdk sdk, we could do it, but it doesn't do it now.
- The way that version is specified on top level sdk's is not using metadata which may not work in case you want to have the same property to control the reference to the SDK and the version of the AppHost package ref.
- To do so now would also make us sort of commit to the fact that AppHost projects will always be of this type of SDK, and it would be really hard if in the future there is a scenario where that is not the case.

Mainly the issue is that while it does look slightly cleaner, it would make us commit to a path that we may want to change in the future and we wouldn't be able to without breaking people again.

---

> To do so now would also make us sort of commit to the fact that AppHost projects will always be of this type of SDK, and it would be really hard if in the future there is a scenario where that is not the case.

> Mainly the issue is that while it does look slightly cleaner, it would make us commit to a path that we may want to change in the future and we wouldn't be able to without breaking people again.

I don't see how the current approach is any different w.r.t. to that aspect. We are embedding the SDK name and version on line 3 instead of line 1. How could we change it without breaking people?

---

cc @baronfel - any input here? Is using the top `<Project Sdk=""` syntax not recommended for a reason like:

> The way that version is specified on top level sdk's is not using metadata

---

For end users if you wanted to manage NuGet-delivered SDK versions uniformly your only real option today is global.json. I'd love to have something more user-friendly here - global.json MSBuild SDK versions aren't exactly integrated with the rest of our tooling, but all of that is owned by the NuGet team entirely.